### PR TITLE
feat(utapi): Add renameFiles to utapi

### PR DESCRIPTION
**Pull request by markflorkowski**
- add UTAPI function
- update type name, add docs

### CODE REVIEW

File: docs/src/pages/api-reference/server.mdx [hunk 1]
```diff
@@ -210,3 +210,41 @@ console.log(files);
...
+```ts
+type Rename = {
+  fileKey: string;
+  newName: string;
+};
+```
+
+| Arguments | Type                 | Required | Description                                |
+| :-------- | :------------------- | :------- | :----------------------------------------- |
+| input     | Rename \|\| Rename[] | Yes      | The fileKey or fileKeys you want to delete |
```
Comment (Hacksore) [inline docs/src/pages/api-reference/server.mdx:250]: The description is copypasta'd here

File: docs/src/pages/api-reference/server.mdx [hunk 2]
```diff
@@ -210,3 +210,41 @@ console.log(files);
  * ]
  */
 ```
+
+### `renameFiles`
+
+`renameFiles` takes in an `Rename` or an array of `Rename`s and renames the
```
Comment (juliusmarminge) [inline docs/src/pages/api-reference/server.mdx:216]:
```suggestion
`renameFiles` takes in a `Rename` or an array of `Rename`s and renames the
```

File: docs/src/pages/api-reference/server.mdx (hunk 1 above)
Comment (juliusmarminge) [inline docs/src/pages/api-reference/server.mdx:250]:
```suggestion
| renames | Rename \|\| Rename[] | Yes | Object specifying what file to rename and the new name |
```

File: packages/uploadthing/src/sdk/index.ts [hunk 3]
```diff
@@ -111,3 +111,53 @@ export const listFiles = async () => {
...
+ * await renameFile([{ fileKey: "2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg", newName: "new_image.jpg" }, { fileKey: "1649353b-04ea-48a2-9db7-31de7f562c8d_image2.jpg", newName: "new_image2.jpg" }]);
+ *
+ */
+
+type Rename = {
+  fileKey: string;
+  newName: string;
+};
+
+export const renameFile = async (updates: Rename | Rename[]) => {
```
Comment (juliusmarminge) [inline packages/uploadthing/src/sdk/index.ts:138]: would it make sense to call this `updateFile` instead for future proofing being able to update other attributes than just the name?

File: packages/uploadthing/src/sdk/index.ts [hunk 4]
```diff
@@ -111,3 +111,53 @@ export const listFiles = async () => {
...
+    headers: {
+      "Content-Type": "application/json",
+      "x-uploadthing-api-key": getApiKeyOrThrow(),
+      "x-uploadthing-version": UPLOADTHING_VERSION,
+    },
+    body: JSON.stringify({ updates }),
+  });
+
+  const json = (await res.json()) as
+    | { files: { key: string; id: string }[] }
```
Comment (juliusmarminge) [inline packages/uploadthing/src/sdk/index.ts:154]: this type does not match what the infra returns

File: packages/uploadthing/src/sdk/index.ts [hunk 5]
```diff
@@ -111,3 +111,53 @@ export const listFiles = async () => {
...
+    | { error: string };
+
+  if (!res.ok || "error" in json) {
+    const message = "error" in json ? json.error : "Unknown error";
+    throw new Error(message);
+  }
+
+  return res.json() as Promise<{ success: boolean }>;
```
Comment (juliusmarminge) [inline packages/uploadthing/src/sdk/index.ts:162]: fix the type above and this cast shouldn't be necessary

**Review by juliusmarminge — CHANGES_REQUESTED** [review]

File: packages/uploadthing/src/sdk/index.ts (hunk 3 above)
Comment (markflorkowski) [reply packages/uploadthing/src/sdk/index.ts:138]: I considered this, but figured we can always add an `updateFile()` in the future, and if all it does now is allow renaming files, it is a bit confusing to name it `updateFile()`

File: docs/src/pages/api-reference/server.mdx [hunk 6]
```diff
@@ -210,3 +210,41 @@ console.log(files);
  * ]
  */
 ```
+
+### `renameFiles`
+
```
Comment (juliusmarminge) [inline docs/src/pages/api-reference/server.mdx:215]: > Added in `v5.3`

**Pull request merged**